### PR TITLE
Implemented UI test for adding 21 filters to role (BZ1277444)

### PR DESCRIPTION
**Pull request by abalakh**
https://bugzilla.redhat.com/show_bug.cgi?id=1277444
```python
py.test tests/foreman/ui/test_role.py -k test_positive_create_with_21_filters
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.2, cov-2.5.1
collected 57 items
2017-09-15 17:26:21 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_role.py .

============================= 56 tests deselected ==============================
================== 1 passed, 56 deselected in 290.98 seconds ===================
```

### CODE REVIEW

**Comment by abalakh** [comment]
Putting DNM tag as this PR depends on #5259 - i've used few fixes to run this test and receive test results, but those fixes should be a part of #5259 refactor. PR needs retesting when #5259 is merged.

File: tests/foreman/ui/test_role.py [hunk 1]
```diff
@@ -339,6 +339,51 @@ def test_positive_create_filter_admin_user_with_orgs(self):
...
+                self.role.add_permission(
+                    role_name,
+                    resource_type=ptype,
+                    permission_list=[perm],
+                )
+            self.assertIsNotNone(
+                self.role.wait_until_element(common_locators['alert.success']))
+            assigned_permissions = self.role.get_permissions(
+                role_name, used_ptypes)
+            self.assertIsNotNone(assigned_permissions)
```
Comment (ldjebran) [inline tests/foreman/ui/test_role.py]: assigned_permissions can never be None, it can be an empty dict

**Comment by abalakh** [comment]
PR rebased, dependency already merged, tests results:
```python
py.test tests/foreman/ui/test_role.py -k test_positive_create_with_21_filters
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.2, cov-2.5.1
collected 57 items
2017-09-27 12:02:42 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_role.py .

============================= 56 tests deselected ==============================
================== 1 passed, 56 deselected in 319.44 seconds ===================
```
Removing DNM tag as PR is ready for review & merge.

**Comment by abalakh** [comment]
DNM again. My repo seems broken, need to fix it and remove unrelated commits from this PR

**Comment by abalakh** [comment]
There we go

**Comment by abalakh** [comment]
@oshtaier that's very good catch, thank you 👍 
Although i'm not quite sure how to address it - pages style is different, it will probably require to return back previous helpers and we'll have 2 ways to fetch permissions, let's say `self.role.filters_get_permissions` and `self.role.get_permissions`. Or what's on your mind?

**Comment by oshtaier** [comment]
It is hard to find best way to do that. Even taking into account that it is two different entities/urls:
https://server/filters?role_id=nn
https://server/roles

**Comment by oshtaier** [comment]
APT

**Comment by oshtaier** [comment]
@ldjebran please review once more

**Comment by ldjebran** [comment]
@abalakh  as you renamed a function used by many tests and when that function was implemented we have run all the TestCase, Please run all the TestCase to be sure nothing broken.

**Comment by ldjebran** [comment]
@oshtaier @abalakh get_permissions is not a workaround it allow to read all permissions as locked roles was implement and we have no other way to read all the permissions (locked and not locked) in the UI with the same workflow.

**Comment by ldjebran** [comment]
so for me we had not to rename that function

**Comment by abalakh** [comment]
@ldjebran i agree for one specific test it'd be easier to implement helper with different name rather than rename original one which will affect numerous tests. 
The thing is original name was wrong, it's not `role.get_permissions`, it's actually `filter.get_permissions` as we're working from different entity page, like @oshtaier pointed it's url is `https://server/filters?role_id=nn` instead of `https://server/roles`. 
And since we don't have `filter` UI entity - prefix is imo the way to go, `filters_get_permissions` sounds not that bad. And we can't name it just `get_permissions` as we actually have a helper which manages permissions from role page which is supposed to be called `role.get_permissions` 🙂 
What's the alternative? Call that role's page helper `role.get_permissions_from_role_page`?

Regarding test results for entire module for simple variable rename - that's definitely overkill, no logic was touched so nothing could change. What's more important in such cases is `git grep` results, as the only reason why new failures can be introduced is accidentally skipped occurrence of old var name.
Results:
```
git grep -n role.get_permissions
tests/foreman/ui/test_role.py:390:            assigned_permissions = self.role.get_permissions(role_name)
(END)

git grep -n role.get_resources
(END)
```
(that 390 line is expected as it's a new test which uses new helper)

**Comment by ldjebran** [comment]
@abalakh you convinced me about the naming, please run the TestCase (preferable) or the minimum tests (optimal) or one test (mandatory) affected by the naming refactoring and will ACK and merge

**Comment by abalakh** [comment]
Test results:
```python
py.test -v tests/foreman/ui/test_role.py::RoleTestCase
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.2, cov-2.5.1
collected 12 items
2017-10-11 13:06:18 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_role.py::RoleTestCase::test_negative_create_with_invalid_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_assign_cloned_role PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_clone_builtin FAILED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_clone_custom PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_filter_admin_user_with_locs PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_filter_admin_user_with_orgs PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_with_21_filters FAILED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_with_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_delete PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_delete_cloned_builtin PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_update_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_update_permission PASSED

==================== 2 failed, 10 passed in 738.36 seconds =====================
```
Re-executed failed tests (that's ok, my env is very unstable and simple mouse move is enough to fail UI test):
```python
py.test -v tests/foreman/ui/test_role.py -k 'test_positive_clone_builtin or test_positive_create_with_21_filters'
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.2, cov-2.5.1
collected 57 items
2017-10-11 13:20:17 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_role.py::RoleTestCase::test_positive_clone_builtin PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_with_21_filters PASSED

============================= 55 tests deselected ==============================
================== 2 passed, 55 deselected in 435.08 seconds ===================
```

**Pull request merged**